Checkout/Checkout/Source/Model/Card+Scheme.swift

```diff
@@ -107,7 +108,7 @@ extension Card {
       case .americanExpress:
         return NSRegularExpression(staticPattern: "^3[47]")
       case .discover:
-        return NSRegularExpression(staticPattern: "^6(011(0[0-9]|[2-4]|74|7[7-9]|8[6-9]|9[0-9])|4[4-9]|5)")
+        return NSRegularExpression(staticPattern: "^65[4-9]|64[4-9]|6011|(622(?:12[6-9]|1[3-9]|[2-8]|9[01]|92[0-5]))")
```

Comment:
please add unit test for new changes. it's hard to know if the new reg exp is right for .discover.

Reply:
I have now added the faulty pan number raised in the ticket